BUILD.md

```diff
@@ -262,17 +262,75 @@ It patches their CMake to correctly import its dependencies.
+
+1. `rippled` binary built with instrumentation data, enabled by the `coverage`
+   option mentioned above
+2. completed run of unit tests, which populates coverage capture data
+3. completed run of `gcovr` tool (which internally invokes either `gcov` or `llvm-cov`)
+   to assemble both instrumentation data and coverage capture data into a coverage report
+
+Above steps are automated into a single target `coverage_report`. The instrumented
+`rippled` binary can be also used for regular development or testing work, at
+the cost of extra disk space utilisation and a small performance hit
+(to store coverage capture). In case of a spurious failure of unit tests, it is
+possile to re-run `coverage_report` target without rebuilding the `rippled` binary
+(since it is simply a dependency of the coverage report target).
+
+The default coverage report format is `html-details`, but the user
+can override it to any of the formats listed in `Builds/CMake/CodeCoverage.cmake`
+by setting `CODE_COVERAGE_REPORT_FORMAT` variable in `cmake`. It is also possible
+to generate more than one format at a time by setting `CODE_COVERAGE_EXTRA_ARGS`
+variable in `cmake`. The specific command line used to run the `gcovr` tool will be
+displayed if `CODE_COVERAGE_VERBOSE` variable is set.
+
+By default, the code coverage tool runs parallel unit tests with `--unittest-jobs`
+ set to the number of available CPU cores. This may cause spurious test
+errors on Apple. Developers can override the number of unit test jobs with
+`CODE_COVERAGE_TEST_PARALLELISM` variable in `cmake`.
+
+Example use with all cmake variables set:
+
+```
+cd .build
+conan install .. --output-folder . --build missing --settings build_type=Debug
+cmake -DCMAKE_BUILD_TYPE=Debug -Dcoverage=ON -DCODE_COVERAGE_REPORT_FORMAT=html-details -DCODE_COVERAGE_EXTRA_ARGS="--json coverage_report.json --json-pretty --txt coverage_report.txt"  -DCODE_COVERAGE_VERBOSE=ON -DCODE_COVERAGE_TEST_PARALLELISM=2 -DCMAKE_TOOLCHAIN_FILE:FILEPATH=build/generators/conan_toolchain.cmake ..
+cmake --build . --target coverage_report
+```
+
+After `coverage_report` target is completed, the generated coverage report will be
+stored inside the build directory, as either of:
+
+- file named `coverage_report.*`, with a suitable extension for the report format, or
+- directory named `coverage_report`, with `index.html` and other files inside, for `html-details` or `html-nested` report formats.
+
+
 ## Options
 
 | Option | Default Value | Description |
 | --- | ---| ---|
 | `assert` | OFF | Enable assertions.
 | `reporting` | OFF | Build the reporting mode feature. |
+| `coverage` | OFF | Prepare the coverage report. |
 | `tests` | ON | Build tests. |
 | `unity` | ON | Configure a unity build. |
 | `san` | N/A | Enable a sanitizer with Clang. Choices are `thread` and `address`. |
 
-[Unity builds][5] may be faster for the first build
+[Unity builds][7] may be faster for the first build
```

Comment:
Existing links should not be relabeled when a new one is added. Not all numbers need to be used. Just add the new link at an unused number. Otherwise the diff gets these frivolous changes.

Reply:
Ok thanks, will change that. I assumed the reader would prefer the links to be in order